src/ert/run_models/model_factory.py

```diff
@@ -280,22 +256,20 @@ def _setup_multiple_data_assimilation(
     restart_run, prior_ensemble = _determine_restart_info(args)
     active_realizations = _validate_num_realizations(args, config)
     return MultipleDataAssimilation(
-        ESMDARunArguments(
-            random_seed=config.random_seed,
-            active_realizations=active_realizations.tolist(),
-            target_ensemble=_iterative_ensemble_format(config, args),
-            weights=args.weights,
-            restart_run=restart_run,
-            prior_ensemble_id=prior_ensemble,
-            starting_iteration=(
-                storage.get_ensemble(prior_ensemble).iteration + 1 if restart_run else 0
-            ),
-            minimum_required_realizations=config.analysis_config.minimum_required_realizations,
-            experiment_name=args.experiment_name,
-        ),
-        config,
-        storage,
-        config.queue_config,
+        random_seed=config.random_seed,
+        active_realizations=active_realizations.tolist(),
+        target_ensemble=_iterative_ensemble_format(config, args),
+        weights=args.weights,
+        restart_run=restart_run,
+        prior_ensemble_id=prior_ensemble,
+        starting_iteration=storage.get_ensemble(prior_ensemble).iteration + 1
+        if restart_run
+        else 0,
```

Comment:
Could we do this prior to the return statement, this is a bit unreadable

Reply:
Did it inside the constructor instead, can move it to model_factory if you prefer, but for me it felt more natual that way.